runtime/sema/checker.go

```diff
@@ -1399,6 +1422,10 @@ func (checker *Checker) recordFunctionDeclarationOrigin(
 	function *ast.FunctionDeclaration,
 	functionType *FunctionType,
 ) *Origin {
+	if !checker.originsAndOccurrencesEnabled {
```

Comment:
Noticed that we actually did a check before using this function, but also do the check inside the function. Is that for extra safety?

Reply:
Yeah, this is just to be defensive